src/Types/IDEntityCadastralNumber.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace AvtoDev\IDEntity\Types;
+
+use Exception;
+use AvtoDev\IDEntity\Helpers\CadastralNumberInfo;
+use AvtoDev\StaticReferences\References\CadastralDistricts\CadastralRegions;
+use AvtoDev\StaticReferences\References\CadastralDistricts\CadastralRegionEntry;
+use AvtoDev\ExtendedLaravelValidator\Extensions\CadastralNumberValidatorExtension;
+
+class IDEntityCadastralNumber extends AbstractTypedIDEntity implements HasCadastralNumberInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getType(): string
+    {
+        return static::ID_TYPE_CADASTRAL_NUMBER;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function normalize($value): ?string
+    {
+        try {
+            // Удаляем все символы, кроме разрешенных (цифры и знак ":")
+            return (string) \preg_replace('~[^\d\:]~u', '', (string) $value);
+        } catch (Exception $e) {
+            return null;
+        }
+    }
+
+    /**
+     * Return parsed fragments of cadastral number.
+     *
+     * @return CadastralNumberInfo
+     */
+    public function getCadastralNumberInfo(): CadastralNumberInfo
+    {
+        return CadastralNumberInfo::parse($this->value);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getRegionData(): ?CadastralRegionEntry
+    {
+        static $regions = null;
+
+        if (! $regions instanceof CadastralRegions) {
+            $regions = new CadastralRegions;
+        }
+
+        $cadastral_number_info = $this->getCadastralNumberInfo();
+
+        return $regions->getRegionByCode($cadastral_number_info->getRegionCode());
```

Comment:
`return $regions->getRegionByCode($this->getCadastralNumberInfo()->getRegionCode());`

Reply:
`changed`